### PR TITLE
fix(helm): refactor monitoring service

### PR DESCRIPTION
**Pull request by nipsufn**
**What type of PR is this?**

fix / refactor monitoring service for envoy-gateway deployment

**What this PR does / why we need it**:

- deployment needs to expose port for it to be accessible via service
- there's no need to expose whole `ports` field from metrics service
- the port on deployment is static as it cannot be adjusted (https://gateway.envoyproxy.io/latest/design/eg-metrics/#envoygatewaymetrics)

**Which issue(s) this PR fixes**:

N/A

### CODE REVIEW

**Comment by arkodg** [comment]
thanks for finding this bug @nipsufn 
should we instead be defining the deployment port here https://github.com/envoyproxy/gateway/blob/72fadb7c1dbe7d660c2900527c34f53cfcfce7ac/charts/gateway-helm/values.tmpl.yaml#L15 ? cc @Xunzhuo

**Comment by nipsufn** [comment]
It looks to me that `.Values.deployment.ports` is used in `envoy-gateway-service.yaml` directly...
If you think about dropping `.Values.envoyGatewayMetricsService` field then I see couple options:
- add `monitoring` port to `.Values.deployment.ports`, drop `envoy-gateway-metrics-service.yaml` and make users scrape metrics against `envoy-gateway-service.yaml`
- add `.Values.deployment.monitoringPorts` and use that for `envoy-gateway-metrics-service.yaml`

I'll also template already present `.Values.deployment.ports` to container spec on the deployment as they are missing.

**Review by zirain — DISMISSED** [review]

File: charts/gateway-helm/templates/envoy-gateway-metrics-service.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" }}
```
Comment (arkodg) [inline charts/gateway-helm/templates/envoy-gateway-metrics-service.yaml:1]: is there any strong reason for having this capabilities condition ?
for the case when this chart is installed before promethus is installed, this service wont get installed, which is not ideal, we ideally dont want to govern the order of install of components, so I'd vote to rm it

Comment (Xunzhuo) [reply charts/gateway-helm/templates/envoy-gateway-metrics-service.yaml:1]: +1

**Comment by zirain** [comment]
please fix DCO

BTW, can you also set prometheus annotations?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**